[re2] re2 port doesn't work with cmake 3.29.1

**Describe the bug**

After GitHub actions runners were updated to use cmake 3.29.1 (https://github.com/actions/runner-images/releases/tag/ubuntu20%2F20240407.1), our builds of ParquetSharp, which depends on re2, have started failing due to this issue in re2: https://github.com/google/re2/issues/488.

This is caused by this change in cmake: https://gitlab.kitware.com/cmake/cmake/-/commit/6ddf8712cdcc1b548af4b502ea91e94cf0bf4f0b

**Environment**

I've reproduced this in my local dev environment:
- OS: Fedora Linux 39
- Compiler: gcc 13.2.1
- Cmake: 3.29.1

**To Reproduce**

Build a project that uses re2:

vcpkg.json:
```
{
  "$schema": "https://raw.githubusercontent.com/microsoft/vcpkg/master/scripts/vcpkg.schema.json",
  "name": "re2-build-test",
  "version-string": "undefined",
  "builtin-baseline": "0bc3f04da6cb8ae1dbd5e2051032c596bb03cffa",
  "dependencies": [
    "re2"
  ]
}
```

CMakeLists.txt:
```
cmake_minimum_required(VERSION 3.16)

project(re2-build-test VERSION 0.1 LANGUAGES CXX)

find_package(re2 CONFIG REQUIRED)

add_executable(program src/main.cc)
target_link_libraries(program PRIVATE re2::re2)
```

src/main.cc:
```
int main() {
  return 0;
}
```

Build with vcpkg:
```
cmake -B build -S . -D CMAKE_TOOLCHAIN_FILE=$VCPKG_ROOT/scripts/buildsystems/vcpkg.cmake
cmake --build build
```

**Expected behavior**

The program should build successfully.

**Failure logs**

Building and installing re2 itself works, but then building the local program fails with:
```
CMake Error at build/vcpkg_installed/x64-linux/share/re2/re2Config.cmake:15 (message):
  File or directory /include referenced by variable re2_INCLUDE_DIR does not                                                                                   
  exist !                                                                      
Call Stack (most recent call first):                                           
  build/vcpkg_installed/x64-linux/share/re2/re2Config.cmake:33 (set_and_check)
  /home/adam/dev/vcpkg/scripts/buildsystems/vcpkg.cmake:859 (_find_package)
  CMakeLists.txt:5 (find_package)   
```

The line causing the error in `build/vcpkg_installed/x64-linux/share/re2/re2Config.cmake` is:
```
set_and_check(re2_INCLUDE_DIR ${PACKAGE_PREFIX_DIR}/include)
```

Because `PACKAGE_PREFIX_DIR` is not defined.

**Additional context**

This has been fixed upstream in re2: https://github.com/google/re2/commit/9ebe4a22cad8a025b68a9594bdff3c047a111333

Would you accept a patch to backport this fix for the current version of re2 in vcpkg?

If not, we can work around this by using our own port of re2 with a patch, or install an older version of cmake on our GitHub action runners, but it seems likely that other users will run into this too.

--- a/ports/re2/portfile.cmake
+++ b/ports/re2/portfile.cmake
@@ -1,9 +1,17 @@
+vcpkg_download_distfile(PATCH_488
+    URLS https://github.com/google/re2/commit/9ebe4a22cad8a025b68a9594bdff3c047a111333.patch?full_index=1
+    SHA512 167a0153bf24e77db2b8555a7d6391acf740dc4720a71e6ce19a3533c7e0a49923d9584ff32544a454837a25eedace00cb2ef9ddbf4250deb5f2c5aeda04ac26
+    FILENAME 9ebe4a22cad8a025b68a9594bdff3c047a111333.patch
+)
+
 vcpkg_from_github(
     OUT_SOURCE_PATH SOURCE_PATH
     REPO google/re2
     REF "${VERSION}"
     SHA512 1511d163ee90c724705cc16d2995e777a7d894ff8133bd3457a26d8c6a9dcb8ccdd2e77b73681e623317a1edbbd3c928569358af91e72ce8612f7b7b61108283
     HEAD_REF master
+    PATCHES
+        "${PATCH_488}"
 )
 
 vcpkg_cmake_configure(

--- a/ports/re2/vcpkg.json
+++ b/ports/re2/vcpkg.json
@@ -1,6 +1,7 @@
 {
   "name": "re2",
   "version-date": "2024-04-01",
+  "port-version": 1,
   "description": "RE2 is a fast, safe, thread-friendly alternative to backtracking regular expression engines like those used in PCRE, Perl, and Python. It is a C++ library.",
   "homepage": "https://github.com/google/re2",
   "license": "BSD-3-Clause",

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -7562,7 +7562,7 @@
     },
     "re2": {
       "baseline": "2024-04-01",
-      "port-version": 0
+      "port-version": 1
     },
     "reactiveplusplus": {
       "baseline": "2.0.0",

--- a/versions/r-/re2.json
+++ b/versions/r-/re2.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "19e35429ba172da1f08743ff317d504fa21bebf3",
+      "version-date": "2024-04-01",
+      "port-version": 1
+    },
+    {
       "git-tree": "aa0f6c4f73231f8b623c799eb8d88e67c6a65481",
       "version-date": "2024-04-01",
       "port-version": 0

